tests/test_result.py

```diff
@@ -193,9 +193,12 @@ def test_binned_mapping():
 
 
 def test_binned_metadata():
+    from heracles.result import Result
+
     md = {"test": object()}
 
     result = np.zeros(3, dtype=np.dtype(float, metadata=md))
+    result = Result(result)
     assert result.dtype.metadata == md
```

Comment:
This should not be necessary here? Otherwise, you have changed the API such that it no longer accepts plain arrays